motr/io_req_fop.c

```diff
@@ -152,11 +152,13 @@ static int application_checksum_process(struct m0_op_io *ioo,
 	cs_data = &irfop->irf_cksum_data;
 	/* Validate if FOP has unit count set */
 	num_units = cs_data->cd_num_units;
-	M0_ASSERT(num_units != 0);
+	if (num_units == 0)
```

Comment:
in all the places in application_checksum_process; it would be better to return failure as if (rc != 0)
			goto fail;
these error cases wont happen ; but would be good to still avoid leak in those rare cases :)

Reply:
done